test/bootloader_test.rb

```diff
@@ -150,4 +158,177 @@
       subject.ReadOrProposeIfNeeded
     end
   end
+
+  describe ".modify_kernel_params" do
+    let(:initial_lines) { {} }
+    let(:params) { { "crashkernel" => "256M" } }
+    let(:append) { "crashkernel=256M" }
+
+    before do
+      Yast::BootCommon.changed = false
+    end
+
+    around do |example|
+      old_globals_value = Yast::BootCommon.globals
+      Yast::BootCommon.globals = initial_lines
+      example.run
+      Yast::BootCommon.globals = old_globals_value
+    end
+
+    context "when no parameters are passed" do
+      it "raises an ArgumentError exception" do
+        expect { subject.modify_kernel_params(:common) }.to raise_error(ArgumentError)
+        expect(Yast::BootCommon.changed).to eq(false)
+      end
+    end
+
+    context "when target does not exist" do
+      it "raises an ArgumentError exception" do
+        expect { subject.modify_kernel_params(:unknown, params) }.to raise_error(ArgumentError)
+        expect(Yast::BootCommon.changed).to eq(false)
+      end
+    end
+
+    context "when no target is specified" do
+      it "uses :common by default" do
+        subject.modify_kernel_params(params)
+
+        expect(Yast::BootCommon.globals)
+          .to eq(kernel_line(:common) => append, "__modified" => "1")
```

Comment:
this is quite fragile, as it can change what is in start in globals, better check directly kernel like key like 
```
expect(Yast::BootCommon.globals[kernel_line(:common)]).to eq append
```